src/main/java/rat/command/RatCommand.java

```diff
@@ -0,0 +1,51 @@
+package rat.command;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+import rat.tasks.RatTaskManager;
+
+/**
+ * This class encapsulates the commands that can be executed by Rat.
+ * Each command has a corresponding 'execute' method that is called by RatInput.
+ */
+public abstract class RatCommand {
+
+    /**
+     * The RatTaskManager object used to store and process the user's tasks.
+     * This RatTaskManager should be initialised in main.
+     */
+    protected final RatTaskManager ratTaskManager;
+
+    /**
+     * Constructor for a RatCommand object.
+     * @param ratTaskManager The RatTaskManager object used to store and process the user's tasks.
+     */
+    protected RatCommand(RatTaskManager ratTaskManager) {
+        this.ratTaskManager = ratTaskManager;
+    }
+
+    /**
+     * Executes the command.
+     */
```

Comment:
What does executing the command do? Should the java docs give more details on what the execute function does?